spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfigurationMockTests.java

```diff
@@ -19,160 +19,141 @@
+	@Test
+	public void testDataSourceProperties() {
+		this.contextRunner.withPropertyValues(
+				"spring.cloud.gcp.sql.instanceConnectionName=proj:reg:test-instance",
+				"spring.datasource.hikari.connectionTestQuery=select 1",
+				"spring.datasource.hikari.maximum-pool-size=19")
+				.run(context -> {
+					HikariDataSource dataSource =
+							(HikariDataSource) context.getBean(DataSource.class);
+					CloudSqlJdbcInfoProvider urlProvider =
+							context.getBean(CloudSqlJdbcInfoProvider.class);
+					assertThat(urlProvider.getJdbcUrl()).isEqualTo(
+							"jdbc:mysql://google/test-database"
+							+ "?cloudSqlInstance=proj:reg:test-instance&socketFactory="
+							+ "com.google.cloud.sql.mysql.SocketFactory&useSSL=false");
+					assertThat(urlProvider.getJdbcDriverClass()).matches("com.mysql.jdbc.Driver");
+					assertThat(dataSource.getMaximumPoolSize()).isEqualTo(19);
+					assertThat(dataSource.getConnectionTestQuery()).matches("select 1");
+				});
 	}
 
-	@TestPropertySource(properties = {
-			"spring.cloud.gcp.sql.instanceConnectionName=world:asia:japan"
-	})
-	public static class GcpCloudSqlAutoConfigurationWithInstanceConnectionNameTest
-			extends GcpCloudSqlAutoConfigurationMockTests {
-		@Test
-		@Override
-		public void test() {
-			assertEquals("jdbc:mysql://google/test-database?cloudSqlInstance=world:asia:japan"
-					+ "&socketFactory=com.google.cloud.sql.mysql.SocketFactory&useSSL=false",
-					this.urlProvider.getJdbcUrl());
-			assertEquals("com.mysql.jdbc.Driver", this.urlProvider.getJdbcDriverClass());
-		}
+	@Test
+	public void testInstanceConnectionName() {
+		this.contextRunner.withPropertyValues(
+				"spring.cloud.gcp.sql.instanceConnectionName=world:asia:japan")
+				.run(context -> {
+					HikariDataSource dataSource =
+							(HikariDataSource) context.getBean(DataSource.class);
+					CloudSqlJdbcInfoProvider urlProvider =
+							context.getBean(CloudSqlJdbcInfoProvider.class);
+					assertThat(urlProvider.getJdbcUrl()).isEqualTo(
+							"jdbc:mysql://google/test-database"
+							+ "?cloudSqlInstance=world:asia:japan&socketFactory="
+							+ "com.google.cloud.sql.mysql.SocketFactory&useSSL=false");
+					assertThat(urlProvider.getJdbcDriverClass()).matches("com.mysql.jdbc.Driver");
+				});
 	}
 
-	@TestPropertySource(properties = {
-			"spring.cloud.gcp.sql.database-type=postgresql",
-			"spring.cloud.gcp.sql.instanceConnectionName=tubular-bells:singapore:test-instance"
-	})
-	public static class GcpCloudSqlAutoConfigurationPostgresTest
-			extends GcpCloudSqlAutoConfigurationMockTests {
-
-		@Test
-		@Override
-		public void test() {
-			assertEquals("jdbc:postgresql://google/test-database?socketFactory="
-					+ "com.google.cloud.sql.postgres.SocketFactory&"
-					+ "socketFactoryArg=tubular-bells:singapore:test-instance&useSSL=false",
-					this.urlProvider.getJdbcUrl());
-			assertEquals("org.postgresql.Driver", this.urlProvider.getJdbcDriverClass());
-		}
-	}
+//	@TestPropertySource(properties = {
```

Comment:
@snicoll is there a way to also control what's in the classpath with `ApplicationContextRunner`? I suspect not, just wanted your confirmation.

One solution to make this test work is to create two separate auto configuration classes that are `@ConditionalOnClass` and add each one separately to each context, but I'd really like to avoid doing that...

Reply:
Yes. You can tune the `ClassLoader` with a `FilteredClassLoader`. See [this example](https://github.com/spring-projects/spring-boot/blob/3f5adfbccc9a624176e17933060003e624bdfaaf/spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfigurationTests.java#L166) that we use to test connection pool detection. You can also filter an individual class.

You can also overwrite the classpath with a more recent version of a jar but that is a rule that applies to the complete test.

Reply:
It is [also available in the latest documentation](https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#boot-features-test-autoconfig)